lib/storage/src/content_manager/collection_meta_ops.rs

```diff
@@ -291,6 +291,11 @@ pub struct ChangeAliasesOperation {
 #[serde(rename_all = "snake_case")]
 pub struct DeleteCollectionOperation(pub String);
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize, Serialize)]
+pub enum ReshardingOperation {
+    Start { peer_id: PeerId, shard_id: ShardId },
```

Comment:
We probably do not want to provide the shard ID here, but select it automagically.

But I guess we can keep it for the time being.

IIRC the shard number **must** be the next one so all are in sequence, otherwise loading shards will fail on restart.

Reply:
This is not exposed to the user of course :man_facepalming: 

I see we have a good place to do it in https://github.com/qdrant/qdrant/pull/4239

Reply:
Yeah, that's the consensus message and user don't have any control over the value of `shard_id` field.